app/assets/javascripts/discourse/app/lib/codeblock-buttons.js

```diff
@@ -0,0 +1,193 @@
+import { cancel, later } from "@ember/runloop";
+import showModal from "discourse/lib/show-modal";
+import I18n from "I18n";
+import { guidFor } from "@ember/object/internals";
+import { clipboardCopy } from "discourse/lib/utilities";
+import { iconHTML } from "discourse-common/lib/icon-library";
+
+// Use to attach copy/fullscreen buttons to a block of code, either
+// within the post stream or for a regular element that contains
+// a pre > code HTML structure.
+//
+// Usage (post):
+//
+// const cb = new CodeblockButtons({
+//   showFullscreen: true,
+//   showCopy: true,
+// });
+// cb.attachToPost(post, postElement);
+//
+// Usage (generic):
+//
+// const cb = new CodeblockButtons({
+//   showFullscreen: true,
+//   showCopy: true,
+// });
+// cb.attachToGeneric(element);
+//
+// Make sure to run .cleanup() on the instance once you are done to
+// remove click events.
+export default class CodeblockButtons {
+  constructor(opts = {}) {
+    this._codeblockButtonClickHandlers = {};
+    this._fadeCopyCodeblocksRunners = {};
+    opts = Object.assign(
+      {
+        showFullscreen: true,
+        showCopy: true,
+      },
+      opts
+    );
+
+    this.showFullscreen = opts.showFullscreen;
+    this.showCopy = opts.showCopy;
+
+    this._boundClickHandler = this._handleClick.bind(this);
+  }
+
+  attachToPost(post, postElement) {
```

Comment:
Is there a chance `attach` can  be called twice on the same element? Would we want to check for existing event listeners prior to adding?

Reply:
I don't think so, nothing in the plugin API says this.